From cc5fff8588125454258c6d45e54fac1b4d72eeda Mon Sep 17 00:00:00 2001
From: Mirza Krak <mirza.krak@hostmobility.com>
Date: Tue, 21 Feb 2017 15:52:28 +0100
Subject: [PATCH 3/3] net: wlcore: add an hard-coded interrupt number

We have no means to pass platform data to the driver, so for now we
hardcode the IRQ number.

NOTE! The irq number is only valid on Colibri T20 and represents
TEGRA_GPIO_PBB2. If SOC or pin is changed this needs to be adjusted.

Signed-off-by: Mirza Krak <mirza.krak@hostmobility.com>
---
 drivers/net/wireless/ti/wlcore/sdio.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ti/wlcore/sdio.c b/drivers/net/wireless/ti/wlcore/sdio.c
index 6946445..a7ceac4 100644
--- a/drivers/net/wireless/ti/wlcore/sdio.c
+++ b/drivers/net/wireless/ti/wlcore/sdio.c
@@ -257,7 +257,14 @@ static int wlcore_probe_of(struct device *dev, int *irq,
 static int wlcore_probe_of(struct device *dev, int *irq,
 			   struct wlcore_platdev_data *pdev_data)
 {
-	return -ENODATA;
+	/* This is a hard-coded value sine we do not yet support to pass
+	   platform data to the driver.
+
+	   410 is the equivalence of TEGRA_GPIO_TO_IRQ(TEGRA_GPIO_PBB2)
+	*/
+	*irq = 410;
+
+	return 0;
 }
 #endif
 
-- 
2.1.4

